src/home/fetch-github/fetch-and-display-previews.ts

```diff
@@ -1,25 +1,33 @@
 import { getImageFromCache } from "../getters/get-indexed-db";
 import { getLocalStore } from "../getters/get-local-store";
+import { GITHUB_TASKS_STORAGE_KEY, TaskStorageItems } from "../github-types";
 import { taskManager } from "../home";
 import { applyAvatarsToIssues, renderGitHubIssues } from "../rendering/render-github-issues";
 import { Sorting } from "../sorting/generate-sorting-buttons";
 import { sortIssuesController } from "../sorting/sort-issues-controller";
 import { fetchAvatar } from "./fetch-avatar";
 import { organizationImageCache } from "./fetch-issues-full";
 import { fetchIssuePreviews } from "./fetch-issues-preview";
-import { TaskMaybeFull, TaskNoFull, TaskNoState, TaskWithFull } from "./preview-to-full-mapping";
+import { TaskMaybeFull, TaskNoFull, TaskWithFull } from "./preview-to-full-mapping";
 
 export type Options = {
   ordering: "normal" | "reverse";
 };
 
 export async function fetchAndDisplayPreviewsFromCache(sorting?: Sorting, options = { ordering: "normal" }) {
-  const _cachedTasks = (getLocalStore("gitHubTasks") || []) as TaskNoState[];
-  const cachedTasks = _cachedTasks.map((task) => ({ ...task, isNew: false, isModified: false })) as TaskMaybeFull[];
+  let _cachedTasks = getLocalStore(GITHUB_TASKS_STORAGE_KEY) as TaskStorageItems;
+  // makes sure tasks have a timestamp to know how old the cache is, or refresh if older than 15 minutes
+  if (!_cachedTasks || !_cachedTasks.timestamp || _cachedTasks.timestamp + 60 * 1000 * 15 <= Date.now()) {
+    _cachedTasks = {
+      timestamp: Date.now(),
+      tasks: [],
+    };
+  }
+  const cachedTasks = _cachedTasks.tasks.map((task) => ({ ...task, isNew: false, isModified: false })) as TaskMaybeFull[];
   taskManager.syncTasks(cachedTasks);
   if (!cachedTasks.length) {
     // load from network if there are no cached issues
-    return await fetchAndDisplayPreviewsFromNetwork(sorting, options);
+    return fetchAndDisplayPreviewsFromNetwork(sorting, options);
```

Comment:
Removing await- is that right?

Reply:
There is nearly no difference between awaiting the return or omitting it. It creates a slightly higher memory use to `return await` because an intermediate Promise is created before the return, and uses one more tick. The only time where it makes a difference is within nested try catch blocks, where scopes of the thrown error will change based on where the await calls are

Reply:
I see so to clarify: it won't cause issues with resolving the promise?

Reply:
Correct! The behavior should be identical.

https://stackoverflow.com/a/42750371/10346405